integration-tests/src/test_loop/tests/congestion_control.rs

```diff
@@ -0,0 +1,176 @@
+use core::panic;
+
+use assert_matches::assert_matches;
+use itertools::Itertools;
+use near_async::test_loop::data::{TestLoopData, TestLoopDataHandle};
+use near_async::test_loop::TestLoopV2;
+use near_async::time::Duration;
+use near_chain_configs::test_genesis::TestGenesisBuilder;
+use near_client::client_actor::ClientActorInner;
+use near_client::Client;
+use near_o11y::testonly::init_test_logger;
+use near_primitives::hash::CryptoHash;
+use near_primitives::types::{AccountId, BlockHeight};
+use near_primitives::views::FinalExecutionStatus;
+
+use crate::test_loop::builder::TestLoopBuilder;
+use crate::test_loop::env::{TestData, TestLoopEnv};
+use crate::test_loop::utils::transactions::{call_contract, deploy_contract, get_node_data};
+use crate::test_loop::utils::ONE_NEAR;
+
+const NUM_PRODUCERS: usize = 2;
+const NUM_VALIDATORS: usize = 2;
+const NUM_RPC: usize = 1;
+const NUM_CLIENTS: usize = NUM_PRODUCERS + NUM_VALIDATORS + NUM_RPC;
+
+/// This test checks that a chunk with too many transactions is rejected by the
+/// chunk validators.
+#[test]
+fn test_congestion_control_adv_chunk_produce() {
```

Comment:
Hm, is this comment and name of the test correct? As far as I can tell, we just deploy a bunch of contracts and call them all.

According to the PR description, this is to exercise state sync with SV enabled and receipt congestion. I don't even understand how state sync is triggered in this but that's probably because I have no idea about the test loop. But I certainly don't understand how this tests adversarial chunks or how this is connected to state sync.

Overall, while the code looks good to me, I'm confused as to what the goal of it is.

Reply:
Ah, the comment is not correct anymore, let me fix that, no more adversarial behaviour. I decided to first add this simple test and then later I'll add the adversarial one based on this. 

In the test loop setup single shard tracking and shard shuffling are enabled. This means that each of the producers only tracks one shard at a time. Since the shard assignment is shuffled every epoch the producers need to state sync the shard for the next epoch.